src/plugins/sync-plugin.js

```diff
@@ -85,58 +137,21 @@ export const ySyncPlugin = (yXmlFragment, { colors = defaultColors, colorMapping
     key: ySyncPluginKey,
     state: {
       init: (initargs, state) => {
-        return {
-          type: yXmlFragment,
-          doc: yXmlFragment.doc,
-          binding: null,
-          snapshot: null,
-          prevSnapshot: null,
-          isChangeOrigin: false,
-          colors,
-          colorMapping,
-          permanentUserData
-        }
+        return pluginState
       },
       apply: (tr, pluginState) => {
-        const change = tr.getMeta(ySyncPluginKey)
-        if (change !== undefined) {
-          pluginState = Object.assign({}, pluginState)
-          for (const key in change) {
-            pluginState[key] = change[key]
-          }
-        }
```

Comment:
I have removed this, I'm unsure of the larger impact, but everything seems to work fine without it in my project?

Reply:
Other plugins might use this to set properties like `snapshot` or `colorMapping`. We can't just remove this.

Reply:
@dmonad I have restored this, but dropped the line `pluginState = Object.assign({}, pluginState)`

I guess this is to make the plugin state immutable? In [the example](https://prosemirror.net/examples/track/) I copied this pattern from, it looks like the plugin apply step returns a new instance of the state to achieve this immutability.

However with the asynchronous setting of the `binding` using the view, I'm not 100% sure how to achieve this, anybody have any suggestions?